src/ui/pages/WalletSelect/WalletSelect.tsx

```diff
@@ -77,18 +78,40 @@ function PortfolioRow({ walletGroups }: { walletGroups: WalletGroup[] }) {
 
 export function WalletSelect() {
   const navigate = useNavigate();
-  const { data: walletGroups, isLoading } = useQuery({
+  const { data: walletGroups, isLoading: isLoadingWalletGroups } = useQuery({
     queryKey: ['wallet/uiGetWalletGroups'],
     queryFn: () => walletPort.request('uiGetWalletGroups'),
     useErrorBoundary: true,
   });
+  const ownedGroups = useMemo(
+    () =>
+      walletGroups?.filter(
+        (group) => !isReadonlyContainer(group.walletContainer)
+      ),
+    [walletGroups]
+  );
+  const ownedAddresses = useMemo(
+    () =>
+      ownedGroups?.flatMap((group) =>
+        group.walletContainer.wallets.map((wallet) => wallet.address)
+      ),
+    [ownedGroups]
+  );
+
+  const { data: walletsMeta, isLoading: isLoadingWalletsMeta } = useQuery({
+    queryKey: ['ZerionAPI.getWalletsMeta', ownedAddresses],
+    queryFn: () =>
+      ownedAddresses ? getWalletsMetaByChunks(ownedAddresses) : null,
+    enabled: Boolean(ownedAddresses),
+  });
 
   const ownedAddressesCount = useMemo(
```

Comment:
Isn't it just a `ownedAddresses.length` now

Reply:
yes :)